x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotRetentionConfiguration.java

```diff
@@ -214,14 +216,14 @@ public Integer getMaximumSnapshotCount() {
                         return false;
                     }
                 }
-
-                if (snapshotAge.compareTo(this.expireAfter) > 0) {
-                    logger.trace("[{}]: ELIGIBLE as snapshot age of {} is older than {}",
-                        snapName, snapshotAge.toHumanReadableString(3), this.expireAfter.toHumanReadableString(3));
+                final long snapshotAge = nowSupplier.getAsLong() - si.startTime();
+                if (snapshotAge > this.expireAfter.getMillis()) {
+                    logger.trace(() -> new ParameterizedMessage("[{}]: ELIGIBLE as snapshot age of [{}ms] is older than {}",
+                        snapName, snapshotAge, this.expireAfter.toHumanReadableString(3)));
                     return true;
                 } else {
-                    logger.trace("[{}]: INELIGIBLE as snapshot age of {} is newer than {}",
-                        snapName, snapshotAge.toHumanReadableString(3), this.expireAfter.toHumanReadableString(3));
+                    logger.trace(() -> new ParameterizedMessage("[{}]: INELIGIBLE as snapshot age of [{}ms] is newer than {}",
+                        snapName, snapshotAge, this.expireAfter.toHumanReadableString(3)));
```

Comment:
Same comment here about including the human-readable age of the snapshot instead of just milliseconds